zygrader/ui/window.py

```diff
@@ -341,22 +345,24 @@ def draw_header(self):
 
         if self.__header_title:
             if callable(self.__header_title):
-                display_text = f"{self.name} | {self.__header_title()}"
+                display_text = f"{self.name} {separator} {self.__header_title()}"
             else:
-                display_text = f"{self.name} | {self.__header_title}"
+                display_text = f"{self.name} {separator} {self.__header_title}"
         else:
             display_text = self.name
 
         if self.__email_text:
-            display_text += f" | {self.__email_text}"
+            display_text += f" {separator} {self.__email_text}"
 
         if self.insert_mode:
-            display_text += " | INSERT"
+            display_text += f" {separator} INSERT"
         elif self.mark_mode:
-            display_text += " | VISUAL"
+            display_text += f" {separator} VISUAL"
 
         # Centered header
         row = self.cols // 2 - len(display_text) // 2
+        if self.spooky_mode:
```

Comment:
Perhaps move this above the comment to separate the logic because `row` is attached to the `add_str`